src/test/librbd/deep_copy/test_mock_ObjectCopyRequest.cc

```diff
@@ -362,8 +362,12 @@ class TestMockDeepCopyObjectCopyRequest : public TestMockFixture {
   }
 
   int create_snap(const char* snap_name) {
+    int r = flush_writeback_cache(m_src_image_ctx);
```

Comment:
We might need to reevaluate the PWL cache's behavior during snapshot creation. If there is no guarantee that the data is consistently persisted when the snapshot is created, that seems like it could lead to massive data corruption:

i.e. create a clone of a new snapshot and it is missing lots of data, or creating a mirroring snapshot and when it performs the deep-copy, it's missing lots of data.

Reply:
Every snapshot creation also issues a request to `block_writes`. In the latest PR, PWL now issues an internal flush when flush source is from `WriteBlockImageDispatch`.